src/alchemlyb/estimators/bar_.py

```diff
@@ -97,18 +97,21 @@ def fit(self, u_nk):
             (len(groups.get_group(i)) if i in groups.groups else 0)
             for i in u_nk.columns
         ]
-
+        states = [x for i, x in enumerate(self._states_) if N_k[i] > 0]
```

Comment:
Would be good to add a test to show why this is needed. So we would always preserve this behaviour.

Reply:
I'm not sure how to test if this is needed or not so changed the way this is defined to more directly pull the lambda states in the indices and throw an error if those lambda states aren't represented in the columns of `u_nk`. This error is now tested with `test_block_average_error_3_bar`